### PR TITLE
Feature #88 - Download leaf ids from branch as text file

### PR DESCRIPTION
**Pull request by jyothishnt**
- Function _downloadLeavesFromBranch()_ to download leaf ids in a branch / all ids
- Added to context menu

### CODE REVIEW

File: src/scripts/PhyloCanvas.js [hunk 1]
```diff
@@ -385,7 +399,17 @@
...
+        handler: 'downloadAllLeaves',
+        internal: false,
+        leaf: false
+      }, {
+        text: 'Download Branch Leaves',
+        handler: 'downloadLeavesFromBranch',
+        internal: true,
+        leaf: false
+      } ];
     }
```
Comment (richardgoater) [inline src/scripts/PhyloCanvas.js]: I'm slightly concerned about the context menu getting quite full of options - I think this feature is a candidate for a plugin or another kind of opt-in in the future, but it's fine for now.

File: src/scripts/PhyloCanvas.js [hunk 2]
```diff
@@ -1487,6 +1509,14 @@
     return this.parent.children.indexOf(this);
   };
 
+  Branch.prototype.downloadLeavesFromBranch = function () {
```
Comment (jyothishnt) [inline src/scripts/PhyloCanvas.js]: Yes. otherwise it will mislead.

File: src/scripts/PhyloCanvas.js [hunk 3]
```diff
@@ -2742,7 +2774,11 @@
       this.historySnapshots = [];
       this.history = new History(this);
     }
-  }
+  };
+
+  Tree.prototype.downloadAllLeaves = function () {
```
Comment (richardgoater) [inline src/scripts/PhyloCanvas.js]: How about updating this to `downloadAllLeafIds`?

Comment (jyothishnt) [inline src/scripts/PhyloCanvas.js]: Sorry mate I forgot :). This is why I need someone to review my pull requests.

Comment (richardgoater) [inline src/scripts/PhyloCanvas.js]: no worries mate, we all need our stuff reviewed for this reason :)

**Comment by richardgoater** [comment]
:+1: Thanks for doing the semicolons :)

**Pull request merged**